src/v/redpanda/tests/fixture.h

```diff
@@ -514,6 +522,17 @@ class redpanda_thread_fixture {
             cluster::without_custom_assignments(std::move(cfgs)),
             model::no_timeout)
           .then([this](std::vector<cluster::topic_result> results) {
+              vassert(
+                results.size() == 1,
+                "expected exactly 1 result but got {}",
+                results.size());
+              const auto& result = results.at(0);
+              if (result.ec != cluster::errc::success) {
+                  throw std::runtime_error(fmt::format(
+                    "error creating topic {}: {}",
```

Comment:
The lack of error handling meant the test would just hang or something like that, if anything failed to be created.

Reply:
nice catrch

Reply:
Yeah caught the hard way by debugging what the heck was going on (some of the BOOST_CHECKPOINT stuff was used for that).